acnportal/acnsim/events/event.py

```diff
@@ -27,6 +31,36 @@ def __lt__(self, other):
         """
         return self.precedence < other.precedence
 
+    @property
+    def type(self):
+        """
+        Legacy accessor for event_type. This will be removed in a future
+        release.
+        """
+        warnings.warn("Accessor 'type' for type of Event is deprecated. "
+                      "Use 'event_type' instead.",
+                      DeprecationWarning)
+        return self.event_type
+
+    def _to_dict(self, context_dict=None):
+        """ Implements BaseSimObj._to_dict. """
+        attribute_dict = {'timestamp': self.timestamp,
+                          'event_type': self.event_type,
+                          'precedence': self.precedence}
+        return attribute_dict, context_dict
+
+    @classmethod
+    def _from_dict_helper(cls, out_obj, attribute_dict):
```

Comment:
these helper functions replace cls_kwargs as a solution to duplicated code.

Reply:
It's more to improve modifiability than for immediate readability. If down the road, we add attributes to Event, only  Event's _from_dict_helper would need to be modified to propagate necessary changes to any Event-like's serializer.